lib/dal/src/job/definition/dependent_values_update.rs

```diff
@@ -144,7 +139,9 @@ impl DependentValuesUpdate {
 
         // Remove the first set of independent_values since they should already have had their functions executed
         for value in dependency_graph.independent_values() {
-            dependency_graph.remove_value(value);
+            if !dependency_graph.values_needs_to_execute_from_prototype_function(value) {
+                dependency_graph.remove_value(value);
+            }
```

Comment:
Okay, I slightly lied. We need to process attribute values that directly dependent on the secret because the secret itself is not an attribute value.